code/_core/obj/structure/interactive/light_switch.dm

```diff
@@ -50,6 +50,9 @@
 	return ..()
 
 /obj/structure/interactive/light_switch/proc/toggle(var/mob/caller)
+	if(!SSgenerator_power.station_power)
+		play_sound('sound/machines/click.ogg',get_turf(src),range_max=VIEW_RANGE*0.5)
+		return TRUE
```

Comment:
I don't know how hard this will be, which is why I'll probably do it if you're not able to, but I think the lightswitch should switch its on-off state but the lights itself don't actually update.

Reply:
I'm not sure I quite understand.